### ISSUE TITLE
Re-creating missing index file won't be finished

### ISSUE THREAD

**Issue by Photographix** [duplicati/duplicati#2820]
- [x ] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info
- **Duplicati version**: v2.0.2.10
- **Operating system**: MacOS 10.12.6
- **Backend**: ACD

## Description
Re-creating missing index file takes a very long time. Won't be finished over days.

## Screenshots
`------------------ BigPool purge ------------------
No certificates found, you can install some with one of these commands:
    cert-sync /etc/ssl/certs/ca-certificates.crt #for Debian based systems
    cert-sync /etc/pki/tls/certs/ca-bundle.crt #for RedHat derivatives
Read more: http://www.mono-project.com/docs/about-mono/releases/3.12.0/#cert-sync
No broken filesets found in database, checking for missing remote files
  Listing remote folder ...
removing file listed as Temporary: duplicati-ia1286f2f15134580ac208b8fa414ca37.dindex.zip.aes
Skipping operation because no files were found to be missing, and no filesets were recorded as broken.
------------------ BigPool repair ------------------
No certificates found, you can install some with one of these commands:
    cert-sync /etc/ssl/certs/ca-certificates.crt #for Debian based systems
    cert-sync /etc/pki/tls/certs/ca-bundle.crt #for RedHat derivatives
Read more: http://www.mono-project.com/docs/about-mono/releases/3.12.0/#cert-sync
  Listing remote folder ...
removing file listed as Temporary: duplicati-ia1286f2f15134580ac208b8fa414ca37.dindex.zip.aes
Destination and database are synchronized, not making any changes
------------------ BigPool BKP ------------------
Backup started at 14.10.2017 21:09:31
No certificates found, you can install some with one of these commands:
    cert-sync /etc/ssl/certs/ca-certificates.crt #for Debian based systems
    cert-sync /etc/pki/tls/certs/ca-bundle.crt #for RedHat derivatives
Read more: http://www.mono-project.com/docs/about-mono/releases/3.12.0/#cert-sync
Checking remote backup ...
  Listing remote folder ...
Re-creating missing index file for duplicati-b2dad4b8b648f44d2bd9350ba17eb1f8e.dblock.zip.aes`

## Debug log
```
2017-10-14 19:09:31Z - Profiling: Starting - Running Backup
2017-10-14 19:09:31Z - Profiling: Starting - Führe ScalarInt64 aus: INSERT INTO "Operation" ("Description", "Timestamp") VALUES (?, ?); SELECT last_insert_rowid();
2017-10-14 19:09:31Z - Profiling: Führe ScalarInt64 aus: INSERT INTO "Operation" ("Description", "Timestamp") VALUES (?, ?); SELECT last_insert_rowid(); took 00:00:00.005
2017-10-14 19:09:31Z - Profiling: Starting - ExecuteReader: SELECT "Key", "Value" FROM "Configuration" 
2017-10-14 19:09:31Z - Profiling: ExecuteReader: SELECT "Key", "Value" FROM "Configuration"  took 00:00:00.000
2017-10-14 19:09:31Z - Profiling: Starting - ExecuteReader: SELECT "Key", "Value" FROM "Configuration" 
2017-10-14 19:09:31Z - Profiling: ExecuteReader: SELECT "Key", "Value" FROM "Configuration"  took 00:00:00.000
2017-10-14 19:09:31Z - Profiling: Starting - Führe ScalarInt64 aus: SELECT COUNT(*) FROM "Block" WHERE "Size" > ?
2017-10-14 19:10:27Z - Profiling: Führe ScalarInt64 aus: SELECT COUNT(*) FROM "Block" WHERE "Size" > ? took 00:00:55.451
2017-10-14 19:10:32Z - Profiling: Starting - ExecuteReader: SELECT "Key", "Value" FROM "Configuration" 
2017-10-14 19:10:32Z - Profiling: ExecuteReader: SELECT "Key", "Value" FROM "Configuration"  took 00:00:00.000
2017-10-14 19:10:32Z - Profiling: Starting - ExecuteReader: SELECT "Key", "Value" FROM "Configuration" 
2017-10-14 19:10:32Z - Profiling: ExecuteReader: SELECT "Key", "Value" FROM "Configuration"  took 00:00:00.000
2017-10-14 19:10:32Z - Profiling: Starting - ExecuteReader: SELECT "CalcLen", "Length", "A"."BlocksetID", "File"."Path" FROM (
SELECT 
    "A"."ID" AS "BlocksetID", 
    IFNULL("B"."CalcLen", 0) AS "CalcLen", 
    "A"."Length"
FROM
    "Blockset" A
LEFT OUTER JOIN
    (
        SELECT 
            "BlocksetEntry"."BlocksetID",
            SUM("Block"."Size") AS "CalcLen"
        FROM
            "BlocksetEntry"
        LEFT OUTER JOIN
            "Block"
        ON
            "Block"."ID" = "BlocksetEntry"."BlockID"
        GROUP BY "BlocksetEntry"."BlocksetID"
    ) B
ON
    "A"."ID" = "B"."BlocksetID"

) A, "File" WHERE "A"."BlocksetID" = "File"."BlocksetID" AND "A"."CalcLen" != "A"."Length" 
2017-10-14 19:12:17Z - Profiling: ExecuteReader: SELECT "CalcLen", "Length", "A"."BlocksetID", "File"."Path" FROM (
SELECT 
    "A"."ID" AS "BlocksetID", 
    IFNULL("B"."CalcLen", 0) AS "CalcLen", 
    "A"."Length"
FROM
    "Blockset" A
LEFT OUTER JOIN
    (
        SELECT 
            "BlocksetEntry"."BlocksetID",
            SUM("Block"."Size") AS "CalcLen"
        FROM
            "BlocksetEntry"
        LEFT OUTER JOIN
            "Block"
        ON
            "Block"."ID" = "BlocksetEntry"."BlockID"
        GROUP BY "BlocksetEntry"."BlocksetID"
    ) B
ON
    "A"."ID" = "B"."BlocksetID"

) A, "File" WHERE "A"."BlocksetID" = "File"."BlocksetID" AND "A"."CalcLen" != "A"."Length"  took 00:01:45.377
2017-10-14 19:12:17Z - Profiling: Starting - Führe ScalarInt64 aus: SELECT Count(*) FROM "BlocklistHash"
2017-10-14 19:12:17Z - Profiling: Führe ScalarInt64 aus: SELECT Count(*) FROM "BlocklistHash" took 00:00:00.225
2017-10-14 19:12:17Z - Profiling: Starting - Führe ScalarInt64 aus: SELECT Count(*) FROM (SELECT DISTINCT "BlocksetID", "Index" FROM "BlocklistHash")
2017-10-14 19:12:18Z - Profiling: Führe ScalarInt64 aus: SELECT Count(*) FROM (SELECT DISTINCT "BlocksetID", "Index" FROM "BlocklistHash") took 00:00:00.113
2017-10-14 19:12:18Z - Profiling: Starting - Führe ScalarInt64 aus: SELECT COUNT(*) FROM (SELECT * FROM (SELECT "N"."BlocksetID", (("N"."BlockCount" + 3200 - 1) / 3200) AS "BlocklistHashCountExpected", CASE WHEN "G"."BlocklistHashCount" IS NULL THEN 0 ELSE "G"."BlocklistHashCount" END AS "BlocklistHashCountActual" FROM (SELECT "BlocksetID", COUNT(*) AS "BlockCount" FROM "BlocksetEntry" GROUP BY "BlocksetID") "N" LEFT OUTER JOIN (SELECT "BlocksetID", COUNT(*) AS "BlocklistHashCount" FROM "BlocklistHash" GROUP BY "BlocksetID") "G" ON "N"."BlocksetID" = "G"."BlocksetID" WHERE "N"."BlockCount" > 1) WHERE "BlocklistHashCountExpected" != "BlocklistHashCountActual")
2017-10-14 19:12:34Z - Profiling: Führe ScalarInt64 aus: SELECT COUNT(*) FROM (SELECT * FROM (SELECT "N"."BlocksetID", (("N"."BlockCount" + 3200 - 1) / 3200) AS "BlocklistHashCountExpected", CASE WHEN "G"."BlocklistHashCount" IS NULL THEN 0 ELSE "G"."BlocklistHashCount" END AS "BlocklistHashCountActual" FROM (SELECT "BlocksetID", COUNT(*) AS "BlockCount" FROM "BlocksetEntry" GROUP BY "BlocksetID") "N" LEFT OUTER JOIN (SELECT "BlocksetID", COUNT(*) AS "BlocklistHashCount" FROM "BlocklistHash" GROUP BY "BlocksetID") "G" ON "N"."BlocksetID" = "G"."BlocksetID" WHERE "N"."BlockCount" > 1) WHERE "BlocklistHashCountExpected" != "BlocklistHashCountActual") took 00:00:16.640
2017-10-14 19:12:34Z - Profiling: Starting - Führe ScalarInt64 aus: SELECT COUNT(*) FROM "File" WHERE "BlocksetID" != ? AND "BlocksetID" != ? AND NOT "BlocksetID" IN (SELECT "BlocksetID" FROM "BlocksetEntry")
2017-10-14 19:12:39Z - Profiling: Führe ScalarInt64 aus: SELECT COUNT(*) FROM "File" WHERE "BlocksetID" != ? AND "BlocksetID" != ? AND NOT "BlocksetID" IN (SELECT "BlocksetID" FROM "BlocksetEntry") took 00:00:05.267
2017-10-14 19:12:39Z - Profiling: Starting - ExecuteReader: SELECT DISTINCT "Fileset"."ID", "Fileset"."Timestamp" FROM "Fileset", "RemoteVolume" WHERE "RemoteVolume"."ID" = "Fileset"."VolumeID" AND "Fileset"."ID" IN (SELECT "FilesetID" FROM "FilesetEntry")  AND ("RemoteVolume"."State" = "Uploading" OR "RemoteVolume"."State" = "Temporary")
2017-10-14 19:12:39Z - Profiling: ExecuteReader: SELECT DISTINCT "Fileset"."ID", "Fileset"."Timestamp" FROM "Fileset", "RemoteVolume" WHERE "RemoteVolume"."ID" = "Fileset"."VolumeID" AND "Fileset"."ID" IN (SELECT "FilesetID" FROM "FilesetEntry")  AND ("RemoteVolume"."State" = "Uploading" OR "RemoteVolume"."State" = "Temporary") took 00:00:00.002
2017-10-14 19:12:39Z - Profiling: Starting - PreBackupVerify
2017-10-14 19:12:39Z - Profiling: Starting - RemoteOperationList
2017-10-14 19:12:39Z - Information: Backend event: List - Started:  ()
2017-10-14 19:12:46Z - Information: Backend event: List - Completed:  (6,90 KB)
2017-10-14 19:12:46Z - Profiling: RemoteOperationList took 00:00:06.748
2017-10-14 19:12:46Z - Profiling: Starting - ExecuteReader: SELECT DISTINCT "Name", "State" FROM "Remotevolume" WHERE "Name" IN (SELECT "Name" FROM "Remotevolume" WHERE "State" IN ("Deleted", "Deleting")) AND NOT "State" IN ("Deleted", "Deleting")
2017-10-14 19:12:47Z - Profiling: ExecuteReader: SELECT DISTINCT "Name", "State" FROM "Remotevolume" WHERE "Name" IN (SELECT "Name" FROM "Remotevolume" WHERE "State" IN ("Deleted", "Deleting")) AND NOT "State" IN ("Deleted", "Deleting") took 00:00:00.043
2017-10-14 19:12:47Z - Profiling: PreBackupVerify took 00:00:07.217
2017-10-14 19:12:47Z - Profiling: Starting - ExecuteReader: SELECT "ID", "Timestamp" FROM "Fileset" ORDER BY "Timestamp" DESC
2017-10-14 19:12:47Z - Profiling: ExecuteReader: SELECT "ID", "Timestamp" FROM "Fileset" ORDER BY "Timestamp" DESC took 00:00:00.000
2017-10-14 19:12:47Z - Profiling: Starting - ExecuteReader: SELECT "Name" FROM "RemoteVolume" WHERE "Type" = ? AND NOT "ID" IN (SELECT "BlockVolumeID" FROM "IndexBlockLink") AND "State" IN (?,?)
2017-10-14 19:12:47Z - Profiling: ExecuteReader: SELECT "Name" FROM "RemoteVolume" WHERE "Type" = ? AND NOT "ID" IN (SELECT "BlockVolumeID" FROM "IndexBlockLink") AND "State" IN (?,?) took 00:00:00.016
2017-10-14 19:12:47Z - Information: Re-creating missing index file for duplicati-b2dad4b8b648f44d2bd9350ba17eb1f8e.dblock.zip.aes
2017-10-14 19:12:47Z - Profiling: Starting - Führe ScalarInt64 aus: INSERT INTO "Remotevolume" ("OperationID", "Name", "Type", "State", "Size", "VerificationCount", "DeleteGraceTime") VALUES (?, ?, ?, ?, ?, ?, ?); SELECT last_insert_rowid();
2017-10-14 19:12:47Z - Profiling: Führe ScalarInt64 aus: INSERT INTO "Remotevolume" ("OperationID", "Name", "Type", "State", "Size", "VerificationCount", "DeleteGraceTime") VALUES (?, ?, ?, ?, ?, ?, ?); SELECT last_insert_rowid(); took 00:00:00.000
2017-10-14 19:12:47Z - Profiling: Starting - ExecuteReader: SELECT "Name", "Hash", "Size" FROM "RemoteVolume" WHERE "Name" = ?
2017-10-14 19:12:47Z - Profiling: ExecuteReader: SELECT "Name", "Hash", "Size" FROM "RemoteVolume" WHERE "Name" = ? took 00:00:00.000
2017-10-14 19:12:47Z - Profiling: Starting - Führe ScalarInt64 aus: SELECT "ID" FROM "Remotevolume" WHERE "Name" = ?
2017-10-14 19:12:47Z - Profiling: Führe ScalarInt64 aus: SELECT "ID" FROM "Remotevolume" WHERE "Name" = ? took 00:00:00.000
2017-10-14 19:12:47Z - Profiling: Starting - ExecuteReader: SELECT DISTINCT "Hash", "Size" FROM "Block" WHERE "VolumeID" = ?
2017-10-14 19:12:47Z - Profiling: ExecuteReader: SELECT DISTINCT "Hash", "Size" FROM "Block" WHERE "VolumeID" = ? took 00:00:00.000
2017-10-14 19:12:47Z - Profiling: Starting - ExecuteReader: SELECT "A"."Hash", "C"."Hash" FROM (SELECT "BlocklistHash"."BlocksetID", "Block"."Hash", * FROM  "BlocklistHash","Block" WHERE  "BlocklistHash"."Hash" = "Block"."Hash" AND "Block"."VolumeID" = ?) A,  "BlocksetEntry" B, "Block" C WHERE "B"."BlocksetID" = "A"."BlocksetID" AND  "B"."Index" >= ("A"."Index" * 3200) AND "B"."Index" < (("A"."Index" + 1) * 3200) AND "C"."ID" = "B"."BlockID"  ORDER BY "A"."BlocksetID", "B"."Index"

```

**Comment by arxcode** [comment]
I just want to confirm that I am having exactly the same issue here (Duplicati 2.0.2.1_beta_2017-08-01 on MacOS X as well, SFTP backend). I am wondering whether this might also be the same issue people are experiencing in #2335.

**Comment by Photographix** [comment]
This problem seems to be fixed or improved in 2.0.12. The recreate for an index file still takes more than a day, but then it goes on.
